src/js/components/productSupplier/create/InvalidItemsIndicator.jsx

```diff
@@ -2,31 +2,79 @@ import React from 'react';
 
 import PropTypes from 'prop-types';
 import { RiCheckboxCircleLine } from 'react-icons/all';
+import { RiErrorWarningLine } from 'react-icons/ri';
 
-import Translate from 'utils/Translate';
-
-const InvalidItemsIndicator = ({ className, errorsCounter }) => (
-  <div className={`invalid-items-indicator ${className}`}>
-    <span>
-      <RiCheckboxCircleLine className="mr-1" />
-      {errorsCounter}
-      {' '}
-      <Translate
-        id="react.productSupplier.form.invalidItemsIndicator.title"
-        defaultMessage="Item(s) require attention"
-      />
-    </span>
-  </div>
-);
+import Button from 'components/form-elements/Button';
+
+const invalidLines = {
+  Icon: RiErrorWarningLine,
+  variant: 'danger',
+  wrapperClassName: 'is-invalid',
+};
+
+const validLines = {
+  Icon: RiCheckboxCircleLine,
+  variant: 'transparent',
+  wrapperClassName: 'is-valid',
+};
+
+const appliedFiltering = ({
+  buttonVariant,
+  isFiltered,
+}) => ({
+  ...buttonVariant,
+  variant: isFiltered ? 'active' : buttonVariant.variant,
+  wrapperClassName: isFiltered ? '' : buttonVariant.wrapperClassName,
+});
+
+const InvalidItemsIndicator = ({
+  className,
+  errorsCounter,
+  setIsFiltered,
+  isFiltered,
+  trigger,
+}) => {
+  const { Icon, variant, wrapperClassName } = appliedFiltering({
+    buttonVariant: errorsCounter ? invalidLines : validLines,
+    isFiltered,
+  });
+
+  const handleOnFilterButtonClick = () => {
+    trigger('productSupplierPreferences');
```

Comment:
Perhaps it would be better to move it to const? Would it be possible to give this function a better name, so it's clear what it triggers?